Plugins/NativeXr/Source/NativeXr.cpp

```diff
@@ -1459,6 +1459,51 @@ namespace Babylon
             XRFrame* m_frame{};
         };
 
+        class XRHand : public Napi::ObjectWrap<XRHand>
+        {
+            static constexpr auto JS_CLASS_NAME = "XRHand";
+
+        public:
+            static void Initialize(Napi::Env env)
+            {
+                Napi::HandleScope scope{env};
+
+                std::vector<Napi::ClassPropertyDescriptor<XRHand>> initList{};
+                initList.reserve(HAND_JOINT_NAMES.size() + 1);
+
+                for (size_t idx = 0; idx < HAND_JOINT_NAMES.size(); idx++)
+                {
+                    initList.push_back(StaticValue(HAND_JOINT_NAMES[idx], Napi::Value::From(env, idx)));
+                }
+
+                initList.push_back(StaticAccessor("length", &XRHand::GetLength, nullptr));
+
+                Napi::Function func = DefineClass(
+                    env,
+                    JS_CLASS_NAME,
+                    initList
```

Comment:
If i remember correctly, this class was previously declared in the webxr types as an array/iteratable item. It looks like we only define the class to have two static fields:

1. N static variables with hand join names that then map to an index value
2. A static variable to get N

Do we need any instance methods on this class for accessing joint data? Or does this class only exist to specify hand joint indices?

Reply:
We would need to specify joint data if this class was being used to pass joint data around as well, but that is being done by the JointSpace vector on InputSource. If the data gets populated elsewhere in the future, like Pose, this would have to be fleshed out, but since all joint setting/getting is done in NativeXR, we can bypass using the class for that, and instead do what we do above, in SetXRInputSourceData(), which essentially creates a NativeXR version of XRHand.

(It's done this way since making a class indexable through N-API is, from what I can tell, impossible. I've played with alternatives, but they aren't nearly as clean)

Reply:
It looks like we may still be missing setting this class up as iterable<XRJointSpace> and an instance accessor that allows querying XRJointSpaces: https://immersive-web.github.io/webxr-hand-input/#xrhand-interface

Is there a reason that this additional functionality isn't implemented?

Reply:
TL:DR the class is only used for static access, and instance access is done through inputSource.HandJoints

Reply:
So we don't have the ability to define the iterable with N-API, is there a reason we aren't defining the getter for a joint?

Reply:
looking at this more it seems like XRHand isn't a type that babylon.js developers will directly interact with/have access to. This seems reasonable for now